tests/test_connection.py

```diff
@@ -1920,6 +1920,7 @@ def test_ssl_not_supported(self):
             self.conn.info.ssl_attribute('wat')
 
     @skip_before_libpq(9, 5)
+    @skip_after_libpq(16)
```

Comment:
I thought you found some probems in 15.1 too? See #1521 

Maybe it would be safer

```suggestion
    @skip_after_libpq(15, 1)
```

or just `15`?